### PR TITLE
[zutils] E: Move Docker flags to build/release subcommands

### LINKED ISSUE #99: Move `--with-docker` from global flag to `build` subcommand flag
## Context

Part of the Windows build/test support effort (#94). This is a prerequisite for the `build` → `test` workflow where build uses Docker and test runs natively.

## Problem

`--with-docker` (and `--with-minimal-docker`, `--docker-image`) is currently a **global flag** parsed before the subcommand:

```bash
./z --with-docker build    # ✅ correct: build inside Docker
./z --with-docker test     # ❌ wrong: wraps test in Docker too
```

This is structurally incorrect. The lifecycle steps have different execution contexts:

| Step | Docker? | KVM? | Notes |
|------|---------|------|-------|
| `setup` | No | No | Downloads sysroot + deps to `.nanvix/` |
| `build` | Yes | No | Cross-compiles inside Docker container |
| `test` | No | Yes | Runs nanvixd.elf natively with `/dev/kvm` |

`--with-docker` sets `self.docker` on the `ZScript` instance, which causes **all** `self.run()` calls to be Docker-wrapped for the entire session. There's no way to say "use Docker for build but not for test." Since each `./z` invocation is a separate process this works in practice today (`./z --with-docker build` then `./z test`), but:

1. The flag position is misleading — it implies it applies to all subcommands
2. `./z --with-docker test` silently does the wrong thing (wraps tests in Docker)
3. It prevents any future single-invocation workflows

## Current code

In `cli.py` (lines 91–113), the Docker flags are on the top-level parser in a mutually exclusive group. In `script.py` `main()` (lines 680–701), they're resolved before the subcommand is dispatched, setting `instance.docker` globally.

## Proposed change

Move `--with-docker`, `--with-minimal-docker`, and `--docker-image` from the top-level parser to the `build` (and optionally `release`) subcommand parsers:

```bash
./z build --with-docker       # ✅ Docker wrapping for build
./z test                      # ✅ Native execution, KVM available
```

### Migration

- Keep the global flags for one release with a deprecation warning: "passing `--with-docker` before the subcommand is deprecated; use `./z build --with-docker` instead"
- After one release, remove the global flags

### Impact on `--all-builds`

`--all-builds` currently passes `docker_image` to `run_all_builds()` which sets `instance.docker` per-combo. This needs to move to per-subcommand handling as well — `--all-builds` only makes sense with `build`, `test`, etc., and Docker should only apply when the subcommand is `build`.

### Impact on consumers

Consumer `z.py` scripts call `self.run()` which checks `self.docker`. After this change, `self.docker` would only be set when the subcommand is `build` (or `release`). Consumer `test()` methods that call `self.run()` would get native execution by default — which is the correct behavior.

## Scope

- `src/nanvix_zutil/cli.py` — move Docker flags to subcommand parsers, add deprecation shim on global parser
- `src/nanvix_zutil/script.py` — move Docker resolution into subcommand dispatch, update `run_all_builds()`
- `tests/test_cli.py` — update flag parsing tests
- `tests/test_script.py` — update Docker dispatch tests

## Acceptance criteria

- `./z build --with-docker` works and wraps build in Docker
- `./z test` runs natively (no Docker wrapping) even if a previous `build` used Docker
- `./z --with-docker build` still works but emits a deprecation warning
- `./z --with-docker test` emits a deprecation warning (and ideally still wraps, for backward compat during migration)
- Full test suite passes, lint/typecheck clean

### PR DESCRIPTION
**Pull request by ada-x64**
## Summary

Moves `--with-docker`, `--with-minimal-docker`, and `--docker-image` from global CLI flags to subcommand-level flags on `build` and `release` only. This prevents `test` from silently running inside Docker (blocking KVM access) and is a prerequisite for Windows build/test support (#94). Closes #99.

Key changes:
- `DEFAULT_DOCKER_SUBCOMMANDS` constant in `cli.py` defines the default set of Docker-eligible subcommands
- `ZScript.DOCKER_SUBCOMMANDS` class attribute references it — consumers override this to expose Docker flags on additional subcommands
- Help pre-parser honours `cls.DOCKER_SUBCOMMANDS` so `--help` output reflects subclass overrides
- `--with-docker` and `--with-minimal-docker` help text clarifies the distinction (overridable default vs fixed minimal image)
- `--docker-image ""` is rejected with a clean error instead of producing a confusing Docker failure
- Integration tests verify the override path end-to-end through `main()`

## Commits

```
53c4be1 [zutils] E: Move Docker flags to subcommands
542f746 [zutils] E: Improve deprecation warning text
3f95621 [zutils] B: Fix deprecated Docker warning text
0177605 [zutils] B: Remove docker deprecation
a9ce04a [zutils] B: Fix Docker CLI edge cases
e8cf8db [zutils] E: Expose DOCKER_SUBCOMMANDS
6ad2202 [zutils] B: Fix Docker CLI review findings
aabd9f1 [tests] E: Test DOCKER_SUBCOMMANDS override
```

### CODE REVIEW

**Comment by ada-x64** [comment]
**A note on rationale:** `--with-docker` is necessary for release as well as build. All downstream repos currently re-build the package before wrapping it.
```python
def release(self) -> None:
    self.run(*self._make_args("package"), cwd=self.repo_root)
    self.run(*self._make_args("verify-package"), cwd=self.repo_root)
```

File: examples/hello-world/.nanvix/z.py [hunk 1]
```diff
@@ -4,14 +4,14 @@
...
-is transparent::
+supported on ``build`` and ``release`` via the ``--with-docker`` /
+``--with-minimal-docker`` / ``--docker-image`` flags; the build script itself
+never references Docker directly — it calls :meth:`~nanvix_zutil.ZScript.run`
+and Docker wrapping is transparent::
 
     nanvix-zutil setup                     # download Nanvix sysroot (host)
     nanvix-zutil build --with-docker       # cross-compile inside Docker container
-    nanvix-zutil test  --with-docker       # run tests (smoke + integration + functional)
+    nanvix-zutil test                      # run tests natively (smoke + integration + functional)
```
Comment (ppenna) [inline examples/hello-world/.nanvix/z.py:11]: Instead of enforcing which command should have this flag, zutils should also provide helper functions for downstream consumers to use docker whenever they feel required. For instance, some packages may need to have a pre-build step before test.
